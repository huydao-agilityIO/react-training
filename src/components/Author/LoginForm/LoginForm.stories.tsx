// LoginForm.stories.ts|tsx

import type { Meta, StoryObj } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { FormProvider, useForm } from 'react-hook-form';

// Components
import LoginForm from './index';

const meta: Meta<typeof LoginForm> = {
  title: 'Components/Author/LoginForm',
  component: LoginForm,
  parameters: {
    layout: 'centered'
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <FormProvider {...useForm()}>
          <Story />
        </FormProvider>
      </MemoryRouter>
    )
  ],
  tags: ['autodocs']
};

export default meta;
type Story = StoryObj<typeof LoginForm>;

export const Default: Story = {
  args: {
    onSubmitForm: () => alert('handles submit form login')
  }
};
