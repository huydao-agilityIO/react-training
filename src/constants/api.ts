export const METHOD = {
  GET: 'GET',
  POST: 'POST',
  PUT: 'PUT',
  DELETE: 'DELETE'
};

export const ERROR_MESSAGES_SERVICES = {
  API: 'API not found!',
  END_POINT: 'Something went wrong!!!'
};

export const END_POINTS = {
  SIGN_IN: 'sign-in',
  SIGN_UP: 'sign-up',
  PATIENT: 'patient',
  HOSPITAL_STAFF: 'hospital-staff'
};

// TODO: Update later
export const API_HOSPITAL_MANAGEMENT = {
  AUTHENTICATION_SIGN_IN: `${process.env.VITE_NX_AUTHENTICATION}/${END_POINTS.SIGN_IN}`,
  AUTHENTICATION_SIGN_UP: `${process.env.VITE_NX_AUTHENTICATION}/${END_POINTS.SIGN_UP}`
};
