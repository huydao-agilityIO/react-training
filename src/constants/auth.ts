export const INVALID_INFO = {
  name: 'Lionel Messi',
  email: 'lionel_messi@co',
  password: 'lio1968'
};

export const VALID_INFO = {
  name: 'Lionel Messi',
  email: 'lionel_messi@gmail.com',
  password: 'lionel_messi1968'
};
