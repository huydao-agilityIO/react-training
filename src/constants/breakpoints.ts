export const breakpoints: Record<string, string> = {
  sm: '414px',
  md: '768px',
  lg: '1024px',
  xl: '1200px',
  '2xl': '1440px'
};
