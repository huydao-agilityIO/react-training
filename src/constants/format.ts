export const CURRENCY_DEFAULT = {
  STYLE: 'currency',
  TEXT: 'USD',
  DECIMAL: 0
};

export const FORMAT_DATE = 'DD/MM/YYYY';
