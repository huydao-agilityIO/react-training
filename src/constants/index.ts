export * from './api';
export * from './auth';
export * from './format';
export * from './patient';
export * from './route';
export * from './sidebar';
export * from './validate';
