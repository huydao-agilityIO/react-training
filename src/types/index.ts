export * from './model';
export * from './sidebar';
export * from './table';
