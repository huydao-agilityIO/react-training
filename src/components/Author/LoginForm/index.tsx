import { Control, Controller } from 'react-hook-form';
import { Button } from '@chakra-ui/react';

// Constants
import { ROUTE } from '@shared/constants';

// Types
import { Authentication } from '@shared/types';

// Utils
import { FORM_SCHEMA } from '@shared/utils/validators';

// Components
import AuthFormContainer from '../AuthFormContainer';
import { FormGroupControl } from '@shared/components';

interface LoginFormProps {
  control?: Control<Omit<Authentication, 'fullName'>>;
  isLoading?: boolean;
  errorResponseAPI?: string;
  onSubmitForm?: () => void;
}

const LoginForm = ({
  control,
  isLoading,
  errorResponseAPI,
  onSubmitForm
}: LoginFormProps) => (
  <AuthFormContainer
    idForm="loginFom"
    mainTitle="Log in"
    linkRouter="Sign Up"
    pathRouter={ROUTE.REGISTER_PAGE}
    errorResponseAPI={errorResponseAPI}
    onSubmit={onSubmitForm}>
    <Controller
      name="email"
      control={control}
      rules={FORM_SCHEMA.EMAIL}
      render={({
        field: { value, onChange, ...rest },
        fieldState: { error }
      }) => {
        return (
          <FormGroupControl
            type="email"
            id="email"
            placeholder="Enter your email"
            value={value}
            errorMessage={error?.message}
            onChange={onChange}
            {...rest}
          />
        );
      }}
    />
    <Controller
      name="password"
      control={control}
      rules={FORM_SCHEMA.PASSWORD}
      render={({
        field: { value, onChange, ...rest },
        fieldState: { error }
      }) => {
        return (
          <FormGroupControl
            type="password"
            id="password"
            placeholder="Password"
            value={value}
            errorMessage={error?.message}
            onChange={onChange}
            {...rest}
          />
        );
      }}
    />
    <Button type="submit" w="full" isLoading={isLoading}>
      Log In
    </Button>
  </AuthFormContainer>
);

export default LoginForm;
