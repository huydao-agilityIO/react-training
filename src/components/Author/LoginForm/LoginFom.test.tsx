import { FormProvider, useForm } from 'react-hook-form';
import { BrowserRouter } from 'react-router-dom';

// Constants
import { VALID_INFO } from '@shared/constants';

// Types
import { Authentication } from '@shared/types';

// Components
import LoginForm from '.';

// TODO: Update later
describe('LoginForm', () => {
  const mockSubmitForm = jest.fn();
  const defaultValues: Partial<Authentication> = {
    email: '',
    password: ''
  };
  const { result } = testLibReactUtils.renderHook(() =>
    useForm<Partial<Authentication>>({ defaultValues })
  );
  const ContainerLoginForm = () => (
    <BrowserRouter>
      <FormProvider {...result.current}>
        <LoginForm onSubmitForm={mockSubmitForm} />
      </FormProvider>
    </BrowserRouter>
  );

  it('renders correctly', () => {
    const { container } = testLibReactUtils.render(<ContainerLoginForm />);

    expect(container).toMatchSnapshot();
  });

  it('submits the form with correct values', () => {
    const { getByPlaceholderText, getByText } = testLibReactUtils.render(
      <ContainerLoginForm />
    );

    testLibReactUtils.fireEvent.change(
      getByPlaceholderText('Enter your email'),
      {
        target: { value: VALID_INFO.email }
      }
    );

    testLibReactUtils.fireEvent.change(getByPlaceholderText('Password'), {
      target: { value: VALID_INFO.password }
    });
    testLibReactUtils.fireEvent.submit(getByText('Log In'));

    expect(mockSubmitForm).toHaveBeenCalled();
  });
});
