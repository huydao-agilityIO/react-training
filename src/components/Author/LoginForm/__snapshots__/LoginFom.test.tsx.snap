// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<div>
  <div
    class="chakra-stack css-1hg7tnx"
  >
    <div
      class="chakra-container css-1s7vqq3"
    >
      <form
        class="css-0"
        id="loginFom"
      >
        <div
          class="chakra-stack css-1e1otyl"
        >
          <div
            class="chakra-stack css-rgfwhv"
          >
            <h2
              class="chakra-heading css-tro5bq"
            >
              Log in
            </h2>
          </div>
          <div
            class="chakra-stack css-feqp41"
          >
            <div
              class="chakra-stack css-9kwbnc"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <input
                  class="chakra-input css-16v8qx8"
                  id="email"
                  placeholder="Enter your email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <input
                  class="chakra-input css-16v8qx8"
                  id="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="chakra-button css-wkgiqm"
                type="submit"
              >
                Log In
              </button>
            </div>
            <div
              class="chakra-stack css-12yeauu"
            >
              <p
                class="chakra-text css-oi2y7z"
              >
                Need an account?
                 
                <a
                  class="chakra-link css-jqihjn"
                  href="/register"
                >
                  Sign Up
                </a>
              </p>
              <a
                class="chakra-link css-3ctcoe"
                href="/"
              >
                Forget Password?
              </a>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
